### ISSUE TITLE
Taskwarrior develop still stores and exports stringified arrays after round trip to Taskserver v1.1.0

### ISSUE THREAD

**Issue by bradyt** [GothenburgBitFactory/taskwarrior#2716]
This issue is a followup to https://github.com/GothenburgBitFactory/taskwarrior/issues/2689. In that issue, there is a long thread that's getting hard to follow, and then a merge request and the issue was closed. As I'm not clear on what was fixed, I'd like to just submit a new issue with what appears to be a prevailing issue, that might be considered a summary duplicate, or a true new issue, after the previous issue was closed.

Overall, I think we have a holistic issue involving Taskwarrior and Taskserver, but the previous issue seemed to be closed with a merge request where we have a reasonable expectation that Taskwarrior develop should be working with Taskserver v1.1.0, so this current issue targets that expectation.

With Taskserver v1.1.0, and Taskwarrior checkout at develop branch, at commit https://github.com/GothenburgBitFactory/taskwarrior/commit/96d6c1df9f084f3f2937895d42c14c25463090b3, I `task import` the following:

```json
[
{"id":1,"description":"foo","entry":"20210101T000000Z","modified":"20210101T000000Z","status":"pending","uuid":"00000000-0000-0000-0000-000000000000","urgency":10},
{"id":2,"description":"bar","entry":"20210101T000000Z","modified":"20210101T000000Z","status":"pending","uuid":"11111111-1111-1111-1111-111111111111","depends":["00000000-0000-0000-0000-000000000000"],"urgency":-3}
]
```

Running `task sync`, the server then stores at `tx.data` the following:

```
{"description":"foo","entry":"20210101T000000Z","modified":"20210101T000000Z","status":"pending","uuid":"00000000-0000-0000-0000-000000000000"}
{"description":"bar","entry":"20210101T000000Z","modified":"20210101T000000Z","status":"pending","uuid":"11111111-1111-1111-1111-111111111111","depends":["00000000-0000-0000-0000-000000000000"]}
22222222-2222-2222-2222-222222222222
```

Setting up a second client, the server sends in response:

```
s: INFO Sending 'XXXXclient: taskd 1.1.0
code: 200
status: Ok

{"description":"foo","entry":"20210101T000000Z","modified":"20210101T000000Z","status":"pending","uuid":"00000000-0000-0000-0000-000000000000"}
{"depends":"[\"00000000-0000-0000-0000-000000000000\"]","description":"bar","entry":"20210101T000000Z","modified":"20210101T000000Z","status":"pending","uuid":"11111111-1111-1111-1111-111111111111"}
22222222-2222-2222-2222-222222222222

' (427 bytes)
```

This client exports the following, where we see the depends field has been stringified:

```json
[
{"id":1,"description":"foo","entry":"20210101T000000Z","modified":"20210101T000000Z","status":"pending","uuid":"00000000-0000-0000-0000-000000000000","urgency":2},
{"id":2,"description":"bar","entry":"20210101T000000Z","modified":"20210101T000000Z","status":"pending","uuid":"11111111-1111-1111-1111-111111111111","depends":["[\"00000000-0000-0000-0000-000000000000\"]"],"urgency":2}
]
```

If it's helpful, here is the contents of `.task/pending.data`:

```
[description:"foo" entry:"1609459200" modified:"1609459200" status:"pending" uuid:"00000000-0000-0000-0000-000000000000"]
[dep_[\"00000000-0000-0000-0000-000000000000\"]:"x" depends:"&open;\\\"00000000-0000-0000-0000-000000000000\\\"&close;" description:"bar" entry:"1609459200" modified:"1609459200" status:"pending" uuid:"11111111-1111-1111-1111-111111111111"]
```

**Comment by djmitche** [comment]
This was part of #2689, ~~and may not be accurate per comments on #2717.~~
[edit: I was thinking of #2714]

**Comment by bradyt** [comment]
I claimed minutes later that I re-verified my findings. (That's me being ~~defensive~~ pedantic for a moment.) But anyways, glad if you consider this duplicate, as I wasn't sure if I misunderstood something.
